README.md

```diff
@@ -1,2 +1,53 @@
 # crates-io-auth-action
-Get a crates.io temporary access token
+
+Get a temporary access token that you can use to interact with crates.io using
+trusted publishing.
+
+## Usage
+
+The action retrieves a token from crates.io and sets it as an output variable.
+You can read the token from the actions output `token` and use it in the subsequent steps.
+The `post` step of the action will revoke the token, so that it cannot be used after the job is done.
+
+To learn how to use the action and configure your crates, read the
+[crates.io documentation](https://crates.io/docs/trusted-publishing).
+
+### Specify a different registry
+
+If you want to use a different registry than crates.io, you can specify the
+registry URL in the `url` input:
+
+```yaml
+- name: Authenticate with custom registry
+  id: auth
+  uses: rust-lang/crates-io-auth-action@v1
+  with:
+    url: my-registry.example.com
+```
+
+Note that the registry must support OpenID Connect authentication.
```

Comment:
in the RFC we explicitly didn't want to make this part of the offical generic registry API protocol, so we might want to restrict this to `staging.crates.io` for now. if this happens to work with other registries too, then that's fine, but I'm not sure if we want to officially support that. if we want this to be truly generic then `crates-io-auth-action` might be the wrong name.

Reply:
I rephrased this 👍